Library/SubscriptionChange.cs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        private SubscriptionAddOnList _addOns;
+        /// <summary>
+        /// The coupon code you want to redeem in the update.
+        /// Only allowed if timeframe is now and you change something about the subscription that creates an invoice.
+        /// </summary>
+        public string CouponCode { get; set; }
+
+        public int? RemainingBillingCycles { get; set; }
+
+        public string CollectionMethod { get; set; }
+
+        public int? NetTerms { get; set; }
+
+        public string PoNumber { get; set; }
+
+        /// <summary>
+        /// Determines whether subscriptions to this plan should auto-renew term at the end of the current term or expire.
+        /// Defaults to true.
+        /// </summary>
+        public bool? AutoRenew { get; set; }
+
+        /// <summary>
+        /// Determines the renewal subscription term.
+        /// Defaults to plans total billing cycles value unless
+        /// overwritten when creating the subscription or editing subscription.
+        /// </summary>
+        public int? RenewalBillingCycles { get; set; }
+
+        public Adjustment.RevenueSchedule? RevenueScheduleType { get; set; }
+
+        /// <summary>
+        /// Optionally set true to indicate if the subscription is to be treated as an import
+        /// for data analysis or a real trial. This value can only be changed on a trial subscription
+        /// and is persisted for its lifetime.
+        /// </summary>
+        public bool? ImportedTrial { get; set; }
+
+        /// <summary>
+        /// Creates a new subscription change object
+        /// </summary>
+        public SubscriptionChange() { }
+
+        #region Write XML documents
+
+        internal void WriteChangeSubscriptionXml(XmlTextWriter xmlWriter)
+        {
+            xmlWriter.WriteStartElement("subscription"); // Start: subscription
+
+            xmlWriter.WriteElementString("timeframe", TimeFrame.ToString().EnumNameToTransportCase());
+
+            if (Quantity.HasValue)
+                xmlWriter.WriteElementString("quantity", Quantity.ToString());
+
+            xmlWriter.WriteStringIfValid("plan_code", PlanCode);
+            xmlWriter.WriteIfCollectionHasAny("subscription_add_ons", AddOns);
+            xmlWriter.WriteStringIfValid("coupon_code", CouponCode);
+
+            if (UnitAmountInCents.HasValue)
+                xmlWriter.WriteElementString("unit_amount_in_cents", UnitAmountInCents.Value.AsString());
+
+            if (CollectionMethod.Like("manual"))
+            {
+                xmlWriter.WriteElementString("collection_method", "manual");
+                xmlWriter.WriteElementString("net_terms", NetTerms.Value.AsString());
+                xmlWriter.WriteElementString("po_number", PoNumber);
+            }
+            else if (CollectionMethod.Like("automatic"))
+                xmlWriter.WriteElementString("collection_method", "automatic");
+
+            if (ImportedTrial.HasValue)
+            {
```

Comment:
Minor nit, but we should try to apply curly braces consistently. Easiest thing to do would probably be to remove them here.